### PR TITLE
Extend generation

### PR DESCRIPTION
**Pull request by hwang-pku**
Fix bugs in master regarding generation and debugging.
Allow for Object key and Object value.
Merge LOG into ConfigTracker.

### CODE REVIEW

**Comment by shuaiwang516** [comment]
This PR still separates the getter and setter tracking and we need to modify all previous instrumented projects?

Also, did you consider the situation that `set()` in ConfuzzGenerator should not be tracked and tagged as SET by test?

**Comment by hwang-pku** [comment]
The new configTracker should be compatible with our previous implementation. I keeped the track method.
I think in ConfuzzGenerator we need generatorSet, which is a set method not tracked by configtracker?

**Comment by shuaiwang516** [comment]
> I think in ConfuzzGenerator we need generatorSet, which is a set method not tracked by configtracker?

Essentially it is achieved by `ConfigTracker.track(key, value, false)`. So you may also need a similar boolean argument

File: confuzz-guidance/src/main/java/edu/illinois/confuzz/internal/ConfuzzGenerator.java [hunk 1]
```diff
@@ -3,27 +3,47 @@
...
+
+        Map<String, Object> configMap = generate(random);
+        Object typedConfigObject = type.cast(configObject);
+        Method setMethod = typedConfigObject.getClass().getMethod(methodName, parameterTypes);
+
+        for (Map.Entry<String, Object> entry: configMap.entrySet()) {
+            try {
+                if (entry.getKey() != null) {
+                    Object key = ConfigTracker.getConfigKeyMap().getOrDefault(entry.getKey(), entry.getKey());
+                    setMethod.invoke(typedConfigObject, key, entry.getValue());
```
Comment (shuaiwang516) [inline confuzz-guidance/src/main/java/edu/illinois/confuzz/internal/ConfuzzGenerator.java:26]: When `generatorSet()` takes two strings as arguments, you may get `java.lang.ClassCastException` here due to Object -> String.

Comment (hwang-pku) [reply confuzz-guidance/src/main/java/edu/illinois/confuzz/internal/ConfuzzGenerator.java:26]: I guess Object -> String is fine as long as value instanceof String. If not, there is a bug either in the way we write our generatorSet or in the way the value is tracked.

**Pull request merged**